src/python/pants/invalidation/build_invalidator.py

```diff
@@ -58,7 +58,9 @@ def __init__(self, cache_key_gen_version=None):
     cache_key_gen_version - If provided, added to all cache keys. Allows you to invalidate
       all cache keys in a single pants repo, by changing this value in config.
     """
-
+    # TODO: Qualify the BuildInvalidator root with the cache_key_gen_version, instead of mixing
+    # it into the key, for uniformity with other qualifiers like the fprint strategy and the
```

Comment:
No, because cache entries in the local/remote cache look like this:
```.buildcache/pants_backend_jvm_tasks_jar_create_JarCreate/pants-plugins.examples.src.java.com.twitter.hello.hello/3b5daf4dad3f6ff625b442ef90ad0280d57597c4-TaskIdentityFingerprintStrategy_8aef9c075871.tgz```